src/ImageSharp/Formats/Tiff/TiffDecoderOptionsParser.cs

```diff
@@ -173,30 +185,39 @@ private static void ParseColorType(this TiffDecoderCore options, ExifProfile exi
 
                 case TiffPhotometricInterpretation.Rgb:
                 {
-                    if (options.BitsPerSample.Bits().Length != 3)
+                    if (options.BitsPerSample.Length != 3)
                     {
                         TiffThrowHelper.ThrowNotSupported("The number of samples in the TIFF BitsPerSample entry is not supported.");
                     }
 
                     if (options.PlanarConfiguration == TiffPlanarConfiguration.Chunky)
                     {
-                        switch (options.BitsPerSample)
+                        ushort bitsPerChannel = options.BitsPerSample[0];
+                        switch (bitsPerChannel)
                         {
-                            case TiffBitsPerSample.Bit42:
+                            case 16:
+                                options.ColorType = TiffColorType.Rgb161616;
```

Comment:
What do you think, is it possible to simplify parsing [TiffColorType](https://github.com/SixLabors/ImageSharp/blob/aba1050bae1a1dc0aee0a72d275933a8f6b41254/src/ImageSharp/Formats/Tiff/PhotometricInterpretation/TiffColorType.cs)? For example, maybe, can move parsing logic directly to [TiffColorDecoderFactory](https://github.com/SixLabors/ImageSharp/blob/aba1050bae1a1dc0aee0a72d275933a8f6b41254/src/ImageSharp/Formats/Tiff/PhotometricInterpretation/TiffColorDecoderFactory%7BTPixel%7D.cs)?

Reply:
while i think this sounds like a sensible idea, I could not come up with a good solution for it. I will leave it now as it is, but maybe we can improve that in the future.